### PR TITLE
Frontend Routing Tweaks

### PR DESCRIPTION
**Pull request by jackblanc**
https://trello.com/c/lJLp7Om9

I also fixed the event props to match the API's structure

### CODE REVIEW

File: src/api/api.js [hunk 1]
```diff
@@ -33,56 +33,40 @@ async function createEvent(event) {
   return res;
 }
 
-const stripePromise = loadStripe(process.env.VUE_APP_STRIPE_PUBLISHABLE_KEY);
+const stripeApp = loadStripe(process.env.VUE_APP_STRIPE_PUBLISHABLE_KEY);
 
-async function handleClickCheckout(cartEvents, userLevel) {
-  if (userLevel === 'ADMIN' || userLevel === 'PF') {
```
Comment (bensenescu) [inline src/api/api.js:39]: Why was this removed?

File: src/api/api.js [hunk 2]
```diff
@@ -33,56 +33,40 @@ async function createEvent(event) {
...
-    const body = {
-      lineItems: cartEvents.map(event => ({
-        id: event.id,
-        name: event.title,
-        description: event.details.description,
-        amount: 5,
-        currency: 'usd',
-        quantity: 1,
-      })),
-      successUrl: `${process.env.VUE_APP_API_DOMAIN}/my-events/?session_id={CHECKOUT_SESSION_ID}`,
```
Comment (bensenescu) [inline src/api/api.js:49]: What is the difference between these two?

File: src/api/api.js [hunk 3]
```diff
@@ -33,56 +33,40 @@ async function createEvent(event) {
...
+      name: event.title,
+      description: event.details.description,
+      amount: 5,
+      currency: 'usd',
+      quantity: 1,
+    })),
+    successUrl: 'http://localhost:8080/my-events',
+    cancelUrl: 'http://localhost:8080/checkout',
+  };
+  const { data } = await protectedResourceAxios.post('/api/v1/protected/checkout/payment', body);
```
Comment (bensenescu) [inline src/api/api.js:65]: So success and failure are just handled in the backend now? all we do is send the data to it?

File: src/api/api.js [hunk 4]
```diff
@@ -33,56 +33,40 @@ async function createEvent(event) {
...
+  const stripe = await stripeApp;
```
Comment (bensenescu) [inline src/api/api.js:66]: function?

Comment (bensenescu) [reply src/api/api.js:66]: Why do you copy it here?

File: src/components/Checkout/PaymentSummary.vue [hunk 5]
```diff
@@ -6,14 +6,12 @@
         <h3>Total: ${{totalPrice.toFixed(2)}}</h3>
         <button
           class="event-btn btn--primary"
-          v-on:click="onClickCheckout">Proceed to checkout</button>
+          @click="clicked">Proceed to checkout</button>
```
Comment (bensenescu) [inline src/components/Checkout/PaymentSummary.vue:9]: v-on:click is preferred

File: src/components/Checkout/PaymentSummary.vue [hunk 6]
```diff
@@ -35,8 +30,8 @@ export default {
     },
   },
   methods: {
-    onClickCheckout() {
-      API.handleClickCheckout(this.registeredEvents, USER[this.adminLevel]);
+    clicked() {
+      this.$emit('onClickCheckout');
```
Comment (bensenescu) [inline src/components/Checkout/PaymentSummary.vue:34]: yurt

File: src/components/Profile/Welcome.vue [hunk 7]
```diff
@@ -2,7 +2,7 @@
   <div class="welcome">
     <img :src="familyImg"/>
     <div>
-      <span class="title">Welcome, the {{ props.name }} Family!</span>
```
Comment (bensenescu) [inline src/components/Profile/Welcome.vue:5]: names<

Comment (bensenescu) [reply src/components/Profile/Welcome.vue:5]: hardcoding>

File: src/components/TheNavigation.vue [hunk 8]
```diff
@@ -1,19 +1,37 @@
 <template>
   <div class="header">
     <div class="title">
-      <router-link to="/" tag="h3">[LOGO]</router-link>
+      <router-link to="/" tag="h3">
+        <img src="https://cdn.firespring.com/images/d8b7f14f-5a80-445d-96e7-49cfd18526f7.png" height="75px"/>
```
Comment (bensenescu) [inline src/components/TheNavigation.vue:5]: dope

Comment (bensenescu) [reply src/components/TheNavigation.vue:5]: maybe store this as a variable logo in data

File: src/components/TheNavigation.vue [hunk 9]
```diff
@@ -1,19 +1,37 @@
...
+      </router-link>
     </div>
-    <div class="navlinks">
-      <router-link to="/about-us" tag="button">About Us</router-link>
-      <router-link to="/events" tag="button">Events</router-link>
-      <access-control :roles="[USER[ROLE.GUEST]]">
+
+
+    <access-control :roles="[USER[ROLE.GUEST]]">
```
Comment (bensenescu) [inline src/components/TheNavigation.vue:10]: Sick refactor

File: src/router/router.js [hunk 10]
```diff
@@ -71,16 +71,14 @@ export default new Router({
       component: SignUpView,
     },
     {
-      path: '/profile/:username',
```
Comment (bensenescu) [inline src/router/router.js:74]: thumbs up

File: src/views/CheckoutView.vue [hunk 11]
```diff
@@ -24,7 +24,9 @@
           </template>
         </events-list-scroll>
       </div>
-      <payment-summary class="payment component-wrapper"></payment-summary>
+      <payment-summary
+        class="payment component-wrapper"
+        @onClickCheckout="onClickCheckout" />
```
Comment (bensenescu) [inline src/views/CheckoutView.vue:29]: v-on

File: src/views/CheckoutView.vue [hunk 12]
```diff
@@ -47,11 +51,27 @@ export default {
...
+      adminLevel: 'adminLevel',
+    }),
   },
   methods: {
     ...mapMutations('cart', {
       cancelRegistration: 'cancelRegistration',
     }),
+    async onClickCheckout() {
+      if (USER[this.adminLevel] === USER[ROLE.ADMIN] || USER[this.adminLevel] === USER[ROLE.PF]) {
+        API.createEventRegistration(this.registeredEvents).then(() => {
```
Comment (bensenescu) [inline src/views/CheckoutView.vue:64]: isn't this route protected anyway

File: src/views/CheckoutView.vue [hunk 13]
```diff
@@ -47,11 +51,27 @@ export default {
...
+          // eslint-disable-next-line
+          alert("Successfully placed order");
+        }).error(() => {
+          // eslint-disable-next-line
+          alert("Error");
```
Comment (bensenescu) [inline src/views/CheckoutView.vue:69]: display the error here

File: src/api/api.js (hunk 1 above)
Comment (jackblanc) [reply src/api/api.js:39]: The logic just got moved to the CheckoutView

File: src/api/api.js (hunk 2 above)
Comment (jackblanc) [reply src/api/api.js:49]: one immediately persists for PF and Admin, one creates a stripe checkout session

File: src/views/CheckoutView.vue (hunk 12 above)
Comment (jackblanc) [reply src/views/CheckoutView.vue:64]: Yeah, we just want to make sure we're calling the right one for the user

File: src/components/TheNavigation.vue (hunk 8 above)
Comment (jackblanc) [reply src/components/TheNavigation.vue:5]: We should do that when we have the logo in our CDN

File: src/App.vue [hunk 14]
```diff
@@ -6,20 +6,21 @@
...
-    <the-footer />
+    <!-- <the-footer /> -->
   </div>
 </template>
 
 <script>
 import TheNavigation from './components/TheNavigation.vue';
 import authService from './utils/service/authService';
-import TheFooter from './components/TheFooter.vue';
+// import TheFooter from './components/TheFooter.vue';
```
[the remark this answers is not in the document]
Comment (jackblanc) [reply src/App.vue:16]: fine

File: src/api/api.js (hunk 4 above)
Comment (jackblanc) [reply src/api/api.js:66]: idk this is just how the demo used it

File: src/api/api.js (hunk 3 above)
Comment (jackblanc) [reply src/api/api.js:65]: unless it fails to create a stripe checkout session, there's no error we surface on the frontend at this step.
